src/cript/nodes/primary_nodes/project.py

```diff
@@ -1,9 +1,173 @@
+from dataclasses import dataclass, field, replace
+from typing import List
+
+from cript.nodes.primary_nodes.collection import Collection
+from cript.nodes.primary_nodes.material import Material
 from cript.nodes.primary_nodes.primary_base_node import PrimaryBaseNode
 
 
 class Project(PrimaryBaseNode):
     """
-    Project node class as described in the data model
+    Project node
     """
 
-    pass
+    @dataclass(frozen=True)
+    class JsonAttributes(PrimaryBaseNode.JsonAttributes):
+        """
+        all Project attributes
+        """
+
+        node: str = "Project"
+        # project name
+        name: str = ""
+        # TODO is group needed?
+        # group: Group = None
+        collections: List[Collection] = field(default_factory=list)
+        materials: List[Material] = field(default_factory=list)
+
+    _json_attrs: JsonAttributes = JsonAttributes()
+
+    def __init__(self, name: str, collections: List[Collection], materials: List[Material] = None, **kwargs):
```

Comment:
> Is collections really required?
> And for a new project the group should be specified.

For the Project node I believe Collection is required because Collection must be nested inside of Project, otherwise how would we associate Project and Collection?

 Also, in the [data model paper](https://pubs.acs.org/doi/suppl/10.1021/acscentsci.3c00011/suppl_file/oc3c00011_si_001.pdf#page=8) it specifies Collection inside of Project

Reply:
That makes all sense, but if I go about creating a new project.
I have to create a collection before the project.
But I can't create a collection, because it requires a project.

So, in my integration test, I passed `[]` for collections in project and retroactively added the collection.
That is unintiutive and error prone.

Reply:
> we need a group argument because that needs to be set somehow.

I originally had that in but I was not sure if that was the limitation of the previous backend or not. 

I think Project could be nested inside of Group and thats it, or does Project need the Group node as well?

could this be what the JSON will look like?

```json
{
"node": "group"
"name": "my group"
    {
        "node": "Project",
        "name": "my project name"
         "collections": []
    }
}
```

Reply:
I created the project first, and add the collections after via the setter functions.

Reply:
> > we need a group argument because that needs to be set somehow.
> 
> I originally had that in but I was not sure if that was the limitation of the previous backend or not.
> 
> I think Project could be nested inside of Group and thats it, or does Project need the Group node as well?
> 
> could this be what the JSON will look like?
> 
> ```json
> {
> "node": "group"
> "name": "my group"
>     {
>         "node": "Project",
>         "name": "my project name"
>          "collections": []
>     }
> }
> ```


This is one of the cycles in the data model.
For me a project needs a group, because that specifies the access permission.
Listing all the projects of a group sounds like a back reference to me.

Reply:
For me it makes sense that Group has a list of Projects inside of it to show that these projects belong to this Group. However, it doesn't make sense for Projects to have Groups inside of them, but I could be misunderstanding things as well. However, I think the safest thing to do is to check with @dylanwal to be sure he agrees.

Reply:
> Overall this looks good and we should merge it soon.
> From there we can probably find more smaller issues via integration testing.

I agree because it's a lot of files and code and chances that there are issues are pretty high, and I am thinking we'll find more issues along the way as we keep testing more.

I am betting the serializations also needs heavier testing, but something we can tackle better after this huge branch is merged in. After that I can try to check every serialization test with the db schema so that it adds another layer of testing and protection